### PR TITLE
ci: spelling: update to 0.0.15a; update whitelist

### PR DESCRIPTION
**Pull request by jsoref**
## Summary of the Pull Request

* Cleaning up the whitelist a bit.

  * The magic to exclude repeated characters worked 👍 

  * Every successful run on master now logs its suggested cleanup, e.g. for https://github.com/microsoft/terminal/commit/5740e197c20bea97b0afe4b396d0541691d02b2c has https://github.com/microsoft/terminal/runs/596271627#step:4:37 

* ⚠️ This check-spelling 0.0.15a+ tolerates Windows line endings in the `whitelist.txt` file (another project I touched had some `.gitconfig` magic which required supporting them).

  This means that if someone edits the file w/ something that likes Windows line endings, the file will successfully convert (instead of it being ignored and check-spelling complaining about everything). Most likely anyone else who then edits the file will use something that will maintain the line endings.

## PR Checklist
* [ ] Closes #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

### CODE REVIEW

File: .github/actions/spell-check/whitelist/whitelist.txt [hunk 1]
```diff
@@ -1555,12 +1517,10 @@ NEWTEXTMETRIC
 NEWTEXTMETRICEX
 Newtonsoft
 NEXTLINE
-nf
 nfe
 nlength
 Nls
 NLSMODE
-nn
 nnn
```
Comment (zadjii-msft) [inline .github/actions/spell-check/whitelist/whitelist.txt:1524]: wait why is `nnn` still needed?

Comment (jsoref) [reply .github/actions/spell-check/whitelist/whitelist.txt:1524]: Looks like it's this:
https://github.com/microsoft/terminal/blob/a13ccfd0f5c1a63ba602d78afba72ab3abaabf34/src/tools/closetest/closetest.cpp#L36
The `:` doesn't fit with the rule. The rule could probably be adjusted. (Or the text changed.)

Comment (jsoref) [reply .github/actions/spell-check/whitelist/whitelist.txt:1524]: (Would `###:` be better?)

Comment (DHowett-MSFT) [reply .github/actions/spell-check/whitelist/whitelist.txt:1524]: I'm fine with either, honestly :smile: ### might be clearer

Comment (jsoref) [reply .github/actions/spell-check/whitelist/whitelist.txt]: I went w/ `###` and I've updated check-spelling.

**Comment by DHowett-MSFT** [comment]
Thanks!

**Pull request merged**